sdk/cosmos/azure-cosmos/pom.xml

```diff
@@ -104,7 +104,12 @@ Licensed under the MIT License.
       <scope>test</scope>
       <version>4.2</version> <!-- {x-version-update;org.apache.commons:commons-collections4;external_dependency} -->
     </dependency>
-
+    <!-- https://mvnrepository.com/artifact/org.hdrhistogram/HdrHistogram -->
+    <dependency>
+      <groupId>org.hdrhistogram</groupId>
+      <artifactId>HdrHistogram</artifactId>
+      <version>2.1.4</version>
```

Comment:
Is this dependency meant for performance tests only? Right now this is being added as a compile time dependency.

Reply:
This is not for test, we are using it to collect telemetry. We have analyzed other options like apache percentile , and this is the more optimize for our use case.

Reply:
@simplynaveen20  - you need to add the tag to this new library added, like this. 

```suggestion
      <version>2.1.4</version> <!-- {x-version-update;cosmos_org.hdrhistogram:HdrHistogram;external_dependency} -->
```

Reply:
@alzimmermsft - do we have any issues with the use of this library ?
Can we go ahead with the use of this ?

Reply:
Adding some more context [Histogram ](https://mvnrepository.com/artifact/org.hdrhistogram/HdrHistogram/2.1.4 )isnot bringing any other library.
And [license ](https://github.com/HdrHistogram/HdrHistogram/blob/master/LICENSE.txt)they are using is "bsd 2-clause license"

Reply:
There isn't a huge issue with on-boarding new compile/runtime dependencies. Generally, taking on new dependencies goes through a review to approve it, https://azure.github.io/azure-sdk/java_implementation.html#dependencies, with a reason on why it is required.

Small question on the dependency itself, why aren't we using the latest version of it? https://search.maven.org/artifact/org.hdrhistogram/HdrHistogram/2.1.12/bundle

cc: @JonathanGiles

Reply:
Thanks @alzimmermsft  for pointing out the lagging version , changed it to latest 2.1.12. I will sync offline for review and approval

Reply:
Got Jonathan's approval over email thread